.github/actions/ci/action.yml

```diff
@@ -0,0 +1,38 @@
+name: CI Workflow
+description: 'Shared CI workflow.'
+inputs:
+  run_tests:
+    description: 'If true, run unit tests, otherwise skip them.'
+    required: false
+    default: 'true'
+  java_version:
+    description: 'The Java version to use.'
+    required: true
+  java_distribution:
+    description: 'The Java distribution to use.'
+    required: false
+    default: 'temurin'
```

Comment:
We don't have any special constraints or considerations, for instance someone may choose to use Corretto because they are good for some specific AWS configurations.

Temurin is very common and is the replacement for AdoptOpenJDK.